cmd/root.go

```diff
@@ -69,14 +68,20 @@ func init() {
 func addCommonParams(cmd *cobra.Command) {
 	// --config / -c a viper configuration file
 	cmd.Flags().StringVarP(&config.CfgFile, "config", "c", "", "config file (default is $HOME/.otel-cli.yaml)")
-
+	// --endpoint an endpoint to send otlp output to
 	cmd.Flags().StringVar(&config.Endpoint, "endpoint", "", "dial address for the desired OTLP/gRPC endpoint")
-	viper.BindPFlag("endpoint", rootCmd.Flags().Lookup("endpoint"))
-	viper.BindEnv("OTEL_EXPORTER_OTLP_ENDPOINT", "endpoint")
-
+	// --timeout a default timeout to use in all otel-cli operations (default 1s)
 	cmd.Flags().StringVar(&config.Timeout, "timeout", "1s", "timeout for otel-cli operations, all timeouts in otel-cli use this value")
-	viper.BindPFlag("timeout", rootCmd.Flags().Lookup("timeout"))
-	viper.BindEnv("OTEL_EXPORTER_OTLP_TIMEOUT", "timeout")
+
+	var common_env_flags = map[string]string{
+		"endpoint": "OTEL_EXPORTER_OTLP_ENDPOINT",
+		"timeout":  "OTEL_EXPORTER_OTLP_TIMEOUT",
+	}
+
+	for config_key, env_value := range common_env_flags {
+		viper.BindPFlag(config_key, cmd.Flags().Lookup(config_key))
+		viper.BindEnv(config_key, env_value)
+	}
```

Comment:
This looks ok but I was also realizing otel-cli startup performance kinda matters so structs might be a bit tighter:

```go
type envFlagMap struct {
    flag string
    env string // or maybe []string? can be a later refactor easily enough
}

// later on...
func whatever() {
    for config_key, env_value := range []envFlagMap{
        { "endpoint", "OTEL_EXPORTER_OTLP_ENDPOINT" },
    } {
        // ...
    }
}